hls-plugin-api/src/Ide/Types.hs

```diff
@@ -206,6 +210,38 @@ data CheckParents
   deriving stock (Eq, Ord, Show, Generic)
   deriving anyclass (FromJSON, ToJSON)
 
+
+data SessionLoadingConfig
+    = SessionLoadSingleComponent
+    -- ^ Always load only a single component when a new component
+    -- is discovered.
+    | SessionLoadMultipleComponents
+    -- ^ Always allow the cradle to load multiple components
+    -- at once. This might not be always possible, if the tool doesn't
+    -- support multiple components loading.
+    --
+    -- The cradle can decide how to handle these situations.
```

Comment:
I think maybe we should just describe these options as _preferences_. i.e. "prefer loading a single component" or whatever, since the implication is that the cradle may do something else if it can't honour it. You could imagine a cradle type that can't honour single component loading (e.g. arguably a direct cradle that loads all components is like that).

Reply:
Updated the comments, will also update the description in vscode-haskell once this PR is merged.